src/Commands/UploadArchive.ts

```diff
@@ -132,6 +132,10 @@ export class UploadArchive implements CommandModule {
       console.log(await archives.upload(spec));
       process.exit(0);
     } catch (e) {
+      if (e.code === 'ERR_STRING_TOO_LONG') {
```

Comment:
This is not a solution to the problem. If we would introduce new upper limits that would not fit to any NodeJS string, we would have the same problem of not being able to upload.
Furthermore, current solution of loading entire file into memory has major drawback especially on low resource machines. 
Archive upload should be reworked to use streams

Reply:
agreed

Reply:
To properly parse and validate the file, it must be loaded into memory, making the use of a stream overkill or even redundant and useless since this limit is propagated to the string as data type affecting reading, JSON serializing, etc. 

The original approach is acceptable, except for the mutation of the error object and message which does not accurately reflects the primary issue. For example, if a user has 2GB of storage space, indicating that the storage is full when a file exceeds 500MB might be misleading. Instead, check the file size during the reading and throw a relevant error message when it exceeds the allowed limit (i.e. 500MB).